src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java

```diff
@@ -0,0 +1,419 @@
+    "000700*   RECORD LENGTH IS 27.                                                  \n" +
+    "000800*                                                                         \n" +
+    "000900        03  DTAR020-KCODE-STORE-KEY.                                      \n" +
+    "001000            05 DTAR020-KEYCODE-NO      PIC X(08).                         \n" +
+    "001100            05 DTAR020-STORE-NO        PIC S9(03)   COMP-3.               \n" +
+    "001200        03  DTAR020-DATE               PIC S9(07)   COMP-3. ")
+  String copybookContents;
+
+  @Name(DROP)
+  @Nullable
+  @Description("Comma-separated list of fields to drop. For example: 'field1,field2,field3'. " +
+    "If both fields to drop and fields to keep are given, fields to keep prevails.")
+  private String drop;
+
+  @Name(KEEP)
+  @Nullable
+  @Description("Comma-separated list of fields to keep. For example: 'field1,field2,field3'. " +
+    "If both fields to drop and fields to keep are given, fields to keep prevails.")
+  private String keep;
+
+  @Name(MAX_SPLIT_SIZE)
+  @Nullable
+  @Description("Maximum split-size(MB) for each mapper in the MapReduce Job. Defaults to 1MB.")
+  @Macro
+  private Long maxSplitSize;
+
+  @Name(CODEPAGE)
+  @Nullable
+  @Description("Code page to use - an alternative notation to the Charset. For example, cp037 or cp322. " +
+    "If this is configured, it overrides the charset property.")
+  @VisibleForTesting
+  String codepage;
+
+  @Name(CHARSET)
+  @Nullable
+  @Description("Charset used to read the data. Available Charset: EBCDIC-US (cp307), EBCDIC-Germany (cp237), " +
+    "EBCDIC-Arabic (cp420), EBCDIC-Denmark, Norway (cp277), EBCDIC-France (cp297),\n" +
+    "EBCDIC-Greece (cp875), EBCDIC-International (cp500), EBCDOC-Italy (cp280), EBCDIC-Russia (cp410)," +
+    " EBCDIC-Spain (cp383), EBCDIC-Thailand (cp838), EBCDIC-Turkey (cp322).")
+  @VisibleForTesting
+  String charset;
+
+  @Name(REPLACEMENTS)
+  @Nullable
+  @Description("List of placeholder strings and their replacements, for example, 'a=b,x=y'. All occurrences of " +
+    "placeholders are replaced prior to loading the copybook. If after replacements a line of the copybook " +
+    "exceeds 72 characters, the copybook will not be accepted and this will result in an error.")
+  @VisibleForTesting
+  String replacements;
+
+  public MainframeSourceConfig(String binaryFilePath, String copybookContents, String drop, String keep,
+                               Long maxSplitSize, String codepage, String charset, String replacements) {
+    super("CopybookReader");
+    this.binaryFilePath = binaryFilePath;
+    this.copybookContents = copybookContents;
+    this.drop = drop;
+    this.keep = keep;
+    this.maxSplitSize = maxSplitSize;
+    this.codepage = codepage;
+    this.charset = charset;
+    this.replacements = replacements;
+  }
+
+  private MainframeSourceConfig(Builder builder) {
+    super("CopybookReader");
```

Comment:
Why are we hardcoding the reference name?

Reply:
It was based implementation. I didn't found reason why should be used hardcoded value.
Updated for using 'referenceName' property.